crates/oxc_semantic/src/jsdoc/builder.rs

```diff
@@ -117,9 +120,7 @@ impl<'a> JSDocBuilder<'a> {
 
         let span = kind.span();
         let mut leading_comments = vec![];
-        // May be better to set range start for perf?
-        // But once I tried, coverage tests start failing...
-        for comment in self.trivias.comments_range(..span.start) {
+        for comment in self.trivias.comments_range(self.previous_span_end..span.start) {
```

Comment:
I think you can delete the `leading_comments` vector as well, and avoid iterating again on it.
It should be enough to conditionally-push directly into `leading_jsdoc_comments`:
```
let comments_len = comments_range.size_hint().1;
let mut leading_jsdoc_comments = Vec::with_capacity(comments_len.unwrap_or(0));

for (_) {
[...]
  self.leading_comments_seen.insert(comment.span.start);
  if let Some(jsdoc) = self.parse_if_jsdoc_comment(_) {
    leading_jsdoc_comments.push(jsdoc);
  }
}

[...]
leading_jsdoc_comments.shrink_to_fit();
self.attached_docs.insert(span, leading_jsdoc_comments);
```

Reply:
Do you wanna give this a try? It's getting late here 😅

Reply:
I think I have to first figure out some issues with my Graphite setup, don't block on me.
I can add this additional change in a followup PR later.

Reply:
You may also push to this branch. Need to wait for a review from the previous author anyway.